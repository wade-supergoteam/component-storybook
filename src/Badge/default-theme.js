// @flow

export const primary = "#007bff";
export const secondary = "#6c757d";
export const success = "#28a745";
export const info = "#17a2b8";
export const warning = "#ffc107";
export const danger = "#dc3545";
export const light = "#f8f9fa";
export const dark = "#343a40";

export const badgeFontSize = "75%";
export const badgeFontWeight = 700;
export const badgePaddingY = ".25em";
export const badgePaddingX = ".4em";
export const badgeBorderRadius = ".25rem";
export const badgePillPaddingX = ".6em";
export const badgePillBorderRadius = "10rem";
