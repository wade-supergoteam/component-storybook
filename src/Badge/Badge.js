// @flow

import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { borderRadius } from "../utils/border-radius";
import themeProp from "../utils/theme";
import * as defaultTheme from "./default-theme";
import { badgeVariant } from "./utils/badge-variant";

const {
  badgeFontSize,
  badgeFontWeight,
  badgePaddingY,
  badgePaddingX,
  badgeBorderRadius,
  badgePillPaddingX,
  badgePillBorderRadius,
} = defaultTheme;

const Badge = styled.span`
  display: inline-block;
  padding: ${themeProp("badgePaddingY", badgePaddingY)} ${themeProp("badgePaddingX", badgePaddingX)};
  font-size: ${themeProp("badgeFontSize", badgeFontSize)};
  font-weight: ${themeProp("badgeFontWeight", badgeFontWeight)};
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  ${borderRadius(themeProp("badgeBorderRadius", badgeBorderRadius))};

  /* Empty badges collapse automatically */
  &:empty {
    display: none;
  }

  ${({ theme, color }) => badgeVariant(theme[color] || defaultTheme[color])};

  ${({ pill }) =>
    pill &&
    css`
      padding-right: ${themeProp("badgePillPaddingX", badgePillPaddingX)};
      padding-left: ${themeProp("badgePillPaddingX", badgePillPaddingX)};
      ${borderRadius(themeProp("badgePillBorderRadius", badgePillBorderRadius))};
    `};
`;

Badge.Link = Badge.withComponent("a");

Badge.propTypes = {
  color: PropTypes.oneOf(["primary", "secondary", "success", "danger", "warning", "info", "light", "dark"]),
  pill: PropTypes.bool,
};

Badge.defaultProps = {
  color: "primary",
  pill: false,
};

export default Badge;
