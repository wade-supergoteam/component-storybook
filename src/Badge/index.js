import Badge from "./Badge";
import { badgeVariant } from "./utils/badge-variant";

export { badgeVariant };
export default Badge;
