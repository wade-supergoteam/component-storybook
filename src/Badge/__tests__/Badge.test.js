import React from "react";
import renderer from "react-test-renderer";
import Badge from "../";

test("it works", () => {
    const wrapper = renderer.create(<Badge />);
    expect(wrapper).toMatchSnapshot();
});

test("Link Badges", () => {
    const wrapper = renderer.create(<Badge Link />);
    expect(wrapper).toMatchSnapshot();
});

test("Pill Badges", () => {
    const wrapper = renderer.create(<Badge pill />);

    expect(wrapper).toMatchSnapshot();
});
