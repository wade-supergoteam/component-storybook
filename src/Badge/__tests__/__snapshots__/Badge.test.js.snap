// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Badges 1`] = `
.c0 {
  display: inline-block;
  padding: .25em .4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: .25rem;
  color: #fff;
  background-color: #007bff;
}

.c0:empty {
  display: none;
}

.c0[href]:hover {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0062cc;
}

<span
  className="c0"
  color="primary"
/>
`;

exports[`Pill Badges 1`] = `
.c0 {
  display: inline-block;
  padding: .25em .4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: .25rem;
  color: #fff;
  background-color: #007bff;
  padding-right: .6em;
  padding-left: .6em;
  border-radius: 10rem;
}

.c0:empty {
  display: none;
}

.c0[href]:hover {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0062cc;
}

<span
  className="c0"
  color="primary"
/>
`;

exports[`it works 1`] = `
.c0 {
  display: inline-block;
  padding: .25em .4em;
  font-size: 75%;
  font-weight: 700;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: .25rem;
  color: #fff;
  background-color: #007bff;
}

.c0:empty {
  display: none;
}

.c0[href]:hover {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #0062cc;
}

<span
  className="c0"
  color="primary"
/>
`;
